apps/election-manager/src/utils/semsTallies.ts

```diff
@@ -429,3 +439,99 @@ function filterTallyForPartyId(
     numberOfBallotsCounted,
   }
 }
+
+export function convertTalliesByPrecinctToSEMSFile(
+  talliesByPrecinct: Dictionary<ExternalTally>,
+  election: Election
+): string {
+  const dataRows: string[][] = []
+  for (const precinctId of Object.keys(talliesByPrecinct)) {
+    const precinctTally = talliesByPrecinct[precinctId]!
+    const contestsForPrecinct = expandEitherNeitherContests(
+      getContestsForPrecinct(election, precinctId)
+    )
+    for (const contest of contestsForPrecinct) {
+      const contestTally = precinctTally.contestTallies[contest.id]
+      const party = contest.partyId
+        ? election.parties.find((p) => p.id === contest.partyId)
+        : undefined
+      const baseRowData = [
+        election.county.id,
+        precinctId,
+        contest.id,
+        contest.title.replace(/\n/g, ''),
+        contest.partyId ?? '0',
+        party?.name ?? 'NP',
+      ]
+
+      // Add rows for under and over votes
+      dataRows.push([
+        ...baseRowData,
+        UndervoteCandidateId,
+        'Times Under Voted',
+        '0',
+        'NP',
+        contestTally?.metadata.undervotes.toString() ?? '0',
+      ])
+      dataRows.push([
+        ...baseRowData,
+        OvervoteCandidateId,
+        'Times Over Voted',
+        '0',
+        'NP',
+        contestTally?.metadata.overvotes.toString() ?? '0',
+      ])
+
+      // For a candidate contest add rows for each candidate and if applicable write ins
+      if (contest.type === 'candidate') {
+        if (contest.allowWriteIns) {
+          dataRows.push([
+            ...baseRowData,
+            WriteInCandidateId,
+            'Write-in',
+            '0',
+            'NP',
+            contestTally?.tallies[writeInCandidate.id]?.tally.toString() ?? '0',
+          ])
+        }
+        for (const candidate of contest.candidates) {
+          const candidateParty = election.parties.find(
+            (p) => p.id === candidate.partyId
+          )
+          dataRows.push([
+            ...baseRowData,
+            candidate.id,
+            candidate.name,
+            candidate.partyId ?? '0',
+            candidateParty ? candidateParty.name : 'NP',
+            contestTally?.tallies[candidate.id]?.tally.toString() ?? '0',
+          ])
+        }
+      }
+
+      // For a yes no contest add yes and no rows
+      if (contest.type === 'yesno') {
+        dataRows.push([
+          ...baseRowData,
+          contest.yesOption!.id,
+          'YES',
+          '0',
+          'NP',
+          contestTally?.tallies.yes?.tally.toString() ?? '0',
+        ])
+        dataRows.push([
+          ...baseRowData,
+          contest.noOption!.id,
+          'NO',
+          '0',
+          'NP',
+          contestTally?.tallies.no?.tally.toString() ?? '0',
+        ])
+      }
+    }
+  }
+  // Transform the data into the expected CSV file format
+  return `${dataRows
+    .map((row) => row.map((e) => `"${e}"`).join(','))
+    .join(',\n')},`
```

Comment:
We need to escape the entries or disallow quotes and possibly more to make this safe.